### PR TITLE
PHOENIX-6234 - Replace Guava with phoenix-thirdparty version in 4.x

### PR DESCRIPTION
**Pull request by gjacoby126**

### CODE REVIEW

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  9s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  7s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 87 new or modified test files.  |
||| _ 4.x Compile Tests _ |
| +0 :ok: |  mvndep  |   5m 13s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  15m 27s |  4.x passed  |
| +1 :green_heart: |  compile  |   1m 53s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   9m 23s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   2m 16s |  4.x passed  |
| +0 :ok: |  spotbugs  |   4m 44s |  root in 4.x has 1010 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   3m 15s |  phoenix-core in 4.x has 958 extant spotbugs warnings.  |
| +0 :ok: |  spotbugs  |   0m 54s |  phoenix-pherf in 4.x has 47 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 30s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  12m 17s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 58s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 58s |  the patch passed  |
| -1 :x: |  checkstyle  |   9m 49s |  root: The patch generated 468 new + 25413 unchanged - 71 fixed = 25881 total (was 25484)  |
| -1 :x: |  whitespace  |   0m  0s |  The patch 6 line(s) with tabs.  |
| +1 :green_heart: |  xml  |   0m  5s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   2m 16s |  the patch passed  |
| -1 :x: |  spotbugs  |   3m 34s |  phoenix-core generated 34 new + 946 unchanged - 12 fixed = 980 total (was 958)  |
| -1 :x: |  spotbugs  |   1m 13s |  phoenix-pherf generated 7 new + 47 unchanged - 0 fixed = 54 total (was 47)  |
| -1 :x: |  spotbugs  |   4m 55s |  root generated 41 new + 998 unchanged - 12 fixed = 1039 total (was 1010)  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 208m 27s |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 32s |  The patch does not generate ASF License warnings.  |
|  |   | 293m  0s |   |


| Reason | Tests |
|-------:|:------|
| FindBugs | module:phoenix-core |
|  |  column must be non-null but is marked as nullable  At MetaDataProtocol.java:is marked as nullable  At MetaDataProtocol.java:[line 215] |
|  |  physicalName must be non-null but is marked as nullable  At MetaDataProtocol.java:is marked as nullable  At MetaDataProtocol.java:[line 222] |
|  |  tableRef must be non-null but is marked as nullable  At MutationState.java:is marked as nullable  At MutationState.java:[line 1970] |
|  |  kv must be non-null but is marked as nullable  At ColumnProjectionFilter.java:is marked as nullable  At ColumnProjectionFilter.java:[lines 151-161] |
|  |  kv must be non-null but is marked as nullable  At EncodedQualifiersColumnProjectionFilter.java:is marked as nullable  At EncodedQualifiersColumnProjectionFilter.java:[lines 117-118] |
|  |  org.apache.phoenix.hbase.index.write.KillServerOnFailurePolicy.handleFailure(Multimap, Exception) invokes System.exit(...), which shuts down the entire virtual machine  At KillServerOnFailurePolicy.java:shuts down the entire virtual machine  At KillServerOnFailurePolicy.java:[line 75] |
|  |  Useless condition:this.disableIndexOnFailure == false at this point  At TrackingParallelWriterIndexCommitter.java:[line 256] |
|  |  input must be non-null but is marked as nullable  At PhoenixIndexFailurePolicy.java:is marked as nullable  At PhoenixIndexFailurePolicy.java:[line 244] |
|  |  entry must be non-null but is marked as nullable  At OrderedResultIterator.java:is marked as nullable  At OrderedResultIterator.java:[line 128] |
|  |  Method org.apache.phoenix.mapreduce.FormatToBytesWritableMapper$1.apply(Object) overrides the nullness annotation of parameter $L1 in an incompatible way  At FormatToBytesWritableMapper.java:annotation of parameter $L1 in an incompatible way  At FormatToBytesWritableMapper.java:[line 370] |
|  |  Method org.apache.phoenix.mapreduce.FormatToBytesWritableMapper$1.apply(String) overrides the nullness annotation of parameter input in an incompatible way  At FormatToBytesWritableMapper.java:annotation of parameter input in an incompatible way  At FormatToBytesWritableMapper.java:[lines 374-380] |
|  |  Possible null pointer dereference in org.apache.phoenix.mapreduce.MultiHfileOutputFormat.configureIncrementalLoad(Job, List) due to return value of called method  Dereferenced at MultiHfileOutputFormat.java:org.apache.phoenix.mapreduce.MultiHfileOutputFormat.configureIncrementalLoad(Job, List) due to return value of called method  Dereferenced at MultiHfileOutputFormat.java:[line 691] |
|  |  Possible null pointer dereference in org.apache.phoenix.mapreduce.MultiHfileOutputFormat.getTableConfigurations(Configuration, String) due to return value of called method  Dereferenced at MultiHfileOutputFormat.java:org.apache.phoenix.mapreduce.MultiHfileOutputFormat.getTableConfigurations(Configuration, String) due to return value of called method  Dereferenced at MultiHfileOutputFormat.java:[line 348] |
|  |  org.apache.phoenix.mapreduce.bulkload.TargetTableRefFunctions.FROM_JSON isn't final but should be  At TargetTableRefFunctions.java:be  At TargetTableRefFunctions.java:[line 48] |
|  |  org.apache.phoenix.mapreduce.bulkload.TargetTableRefFunctions.LOGICAL_NAMES_TO_JSON isn't final but should be  At TargetTableRefFunctions.java:be  At TargetTableRefFunctions.java:[line 78] |
|  |  org.apache.phoenix.mapreduce.bulkload.TargetTableRefFunctions.NAMES_FROM_JSON isn't final but should be  At TargetTableRefFunctions.java:be  At TargetTableRefFunctions.java:[line 95] |
|  |  org.apache.phoenix.mapreduce.bulkload.TargetTableRefFunctions.NAMES_TO_JSON isn't final but should be  At TargetTableRefFunctions.java:be  At TargetTableRefFunctions.java:[line 61] |
|  |  org.apache.phoenix.mapreduce.bulkload.TargetTableRefFunctions.TO_JSON isn't final but should be  At TargetTableRefFunctions.java:be  At TargetTableRefFunctions.java:[line 35] |
|  |  qCol must be non-null but is marked as nullable  At IndexColumnNames.java:is marked as nullable  At IndexColumnNames.java:[line 161] |
|  |  Method org.apache.phoenix.schema.ArgumentTypeMismatchException$1.apply(Class) overrides the nullness annotation of parameter input in an incompatible way  At ArgumentTypeMismatchException.java:annotation of parameter input in an incompatible way  At ArgumentTypeMismatchException.java:[line 51] |
|  |  Method org.apache.phoenix.schema.ArgumentTypeMismatchException$1.apply(Object) overrides the nullness annotation of parameter $L1 in an incompatible way  At ArgumentTypeMismatchException.java:annotation of parameter $L1 in an incompatible way  At ArgumentTypeMismatchException.java:[line 48] |
|  |  Dead store to metaProperties in org.apache.phoenix.schema.MetaDataClient.addColumn(PTable, List, ListMultimap, boolean, boolean, NamedTableNode, PTableType, boolean, List)  At MetaDataClient.java:org.apache.phoenix.schema.MetaDataClient.addColumn(PTable, List, ListMultimap, boolean, boolean, NamedTableNode, PTableType, boolean, List)  At MetaDataClient.java:[line 3896] |
|  |  Found reliance on default encoding in org.apache.phoenix.schema.MetaDataClient.addColumn(PTable, List, ListMultimap, boolean, boolean, NamedTableNode, PTableType, boolean, List):in org.apache.phoenix.schema.MetaDataClient.addColumn(PTable, List, ListMultimap, boolean, boolean, NamedTableNode, PTableType, boolean, List): String.getBytes()  At MetaDataClient.java:[line 4157] |
|  |  Possible null pointer dereference of colDef in org.apache.phoenix.schema.MetaDataClient.addColumn(PTable, List, ListMultimap, boolean, boolean, NamedTableNode, PTableType, boolean, List)  Dereferenced at MetaDataClient.java:colDef in org.apache.phoenix.schema.MetaDataClient.addColumn(PTable, List, ListMultimap, boolean, boolean, NamedTableNode, PTableType, boolean, List)  Dereferenced at MetaDataClient.java:[line 3976] |
|  |  org.apache.phoenix.schema.MetaDataClient.addColumn(PTable, List, ListMultimap, boolean, boolean, NamedTableNode, PTableType, boolean, List) may fail to close PreparedStatement  At MetaDataClient.java:NamedTableNode, PTableType, boolean, List) may fail to close PreparedStatement  At MetaDataClient.java:[line 4133] |
|  |  Nullcheck of newIndexes at line 190 of value previously dereferenced in org.apache.phoenix.schema.PMetaDataImpl.removeTable(PName, String, String, long)  At PMetaDataImpl.java:190 of value previously dereferenced in org.apache.phoenix.schema.PMetaDataImpl.removeTable(PName, String, String, long)  At PMetaDataImpl.java:[line 190] |
|  |  Method org.apache.phoenix.util.QueryUtil$2.apply(Object) overrides the nullness annotation of parameter $L1 in an incompatible way  At QueryUtil.java:annotation of parameter $L1 in an incompatible way  At QueryUtil.java:[line 220] |
|  |  Method org.apache.phoenix.util.QueryUtil$2.apply(String) overrides the nullness annotation of parameter columnName in an incompatible way  At QueryUtil.java:annotation of parameter columnName in an incompatible way  At QueryUtil.java:[line 224] |
|  |  Method org.apache.phoenix.util.csv.CsvUpsertExecutor$ArrayDatatypeConversionFunction.apply(Object) overrides the nullness annotation of parameter $L1 in an incompatible way  At CsvUpsertExecutor.java:annotation of parameter $L1 in an incompatible way  At CsvUpsertExecutor.java:[line 217] |
|  |  Method org.apache.phoenix.util.csv.CsvUpsertExecutor$ArrayDatatypeConversionFunction.apply(String) overrides the nullness annotation of parameter input in an incompatible way  At CsvUpsertExecutor.java:annotation of parameter input in an incompatible way  At CsvUpsertExecutor.java:[lines 229-231] |
|  |  Method org.apache.phoenix.util.csv.CsvUpsertExecutor$SimpleDatatypeConversionFunction.apply(Object) overrides the nullness annotation of parameter $L1 in an incompatible way  At CsvUpsertExecutor.java:annotation of parameter $L1 in an incompatible way  At CsvUpsertExecutor.java:[line 123] |
|  |  Method org.apache.phoenix.util.csv.CsvUpsertExecutor$SimpleDatatypeConversionFunction.apply(String) overrides the nullness annotation of parameter input in an incompatible way  At CsvUpsertExecutor.java:annotation of parameter input in an incompatible way  At CsvUpsertExecutor.java:[lines 168-210] |
|  |  Method org.apache.phoenix.util.json.JsonUpsertExecutor$ArrayDatatypeConversionFunction.apply(Object) overrides the nullness annotation of parameter input in an incompatible way  At JsonUpsertExecutor.java:annotation of parameter input in an incompatible way  At JsonUpsertExecutor.java:[lines 251-253] |
|  |  Method org.apache.phoenix.util.json.JsonUpsertExecutor$SimpleDatatypeConversionFunction.apply(Object) overrides the nullness annotation of parameter input in an incompatible way  At JsonUpsertExecutor.java:annotation of parameter input in an incompatible way  At JsonUpsertExecutor.java:[lines 186-232] |
| FindBugs | module:phoenix-pherf |
|  |  Possible null pointer dereference in org.apache.phoenix.pherf.workload.mt.handlers.RendezvousingWorkHandler.onEvent(BaseLoadEventGenerator$TenantOperationEvent) due to return value of called method  Dereferenced at RendezvousingWorkHandler.java:org.apache.phoenix.pherf.workload.mt.handlers.RendezvousingWorkHandler.onEvent(BaseLoadEventGenerator$TenantOperationEvent) due to return value of called method  Dereferenced at RendezvousingWorkHandler.java:[line 76] |
|  |  Possible null pointer dereference in org.apache.phoenix.pherf.workload.mt.handlers.TenantOperationWorkHandler.onEvent(BaseLoadEventGenerator$TenantOperationEvent) due to return value of called method  Dereferenced at TenantOperationWorkHandler.java:org.apache.phoenix.pherf.workload.mt.handlers.TenantOperationWorkHandler.onEvent(BaseLoadEventGenerator$TenantOperationEvent) due to return value of called method  Dereferenced at TenantOperationWorkHandler.java:[line 65] |
|  |  input must be non-null but is marked as nullable  At IdleTimeOperationSupplier.java:is marked as nullable  At IdleTimeOperationSupplier.java:[lines 51-74] |
|  |  input must be non-null but is marked as nullable  At PreScenarioOperationSupplier.java:is marked as nullable  At PreScenarioOperationSupplier.java:[lines 51-85] |
|  |  input must be non-null but is marked as nullable  At QueryOperationSupplier.java:is marked as nullable  At QueryOperationSupplier.java:[lines 54-94] |
|  |  input must be non-null but is marked as nullable  At UpsertOperationSupplier.java:is marked as nullable  At UpsertOperationSupplier.java:[lines 58-161] |
|  |  input must be non-null but is marked as nullable  At UserDefinedOperationSupplier.java:is marked as nullable  At UserDefinedOperationSupplier.java:[lines 43-47] |
| FindBugs | module:root |
|  |  column must be non-null but is marked as nullable  At MetaDataProtocol.java:is marked as nullable  At MetaDataProtocol.java:[line 215] |
|  |  physicalName must be non-null but is marked as nullable  At MetaDataProtocol.java:is marked as nullable  At MetaDataProtocol.java:[line 222] |
|  |  tableRef must be non-null but is marked as nullable  At MutationState.java:is marked as nullable  At MutationState.java:[line 1970] |
|  |  kv must be non-null but is marked as nullable  At ColumnProjectionFilter.java:is marked as nullable  At ColumnProjectionFilter.java:[lines 151-161] |
|  |  kv must be non-null but is marked as nullable  At EncodedQualifiersColumnProjectionFilter.java:is marked as nullable  At EncodedQualifiersColumnProjectionFilter.java:[lines 117-118] |
|  |  org.apache.phoenix.hbase.index.write.KillServerOnFailurePolicy.handleFailure(Multimap, Exception) invokes System.exit(...), which shuts down the entire virtual machine  At KillServerOnFailurePolicy.java:shuts down the entire virtual machine  At KillServerOnFailurePolicy.java:[line 75] |
|  |  Useless condition:this.disableIndexOnFailure == false at this point  At TrackingParallelWriterIndexCommitter.java:[line 256] |
|  |  input must be non-null but is marked as nullable  At PhoenixIndexFailurePolicy.java:is marked as nullable  At PhoenixIndexFailurePolicy.java:[line 244] |
|  |  entry must be non-null but is marked as nullable  At OrderedResultIterator.java:is marked as nullable  At OrderedResultIterator.java:[line 128] |
|  |  Method org.apache.phoenix.mapreduce.FormatToBytesWritableMapper$1.apply(Object) overrides the nullness annotation of parameter $L1 in an incompatible way  At FormatToBytesWritableMapper.java:annotation of parameter $L1 in an incompatible way  At FormatToBytesWritableMapper.java:[line 370] |
|  |  Method org.apache.phoenix.mapreduce.FormatToBytesWritableMapper$1.apply(String) overrides the nullness annotation of parameter input in an incompatible way  At FormatToBytesWritableMapper.java:annotation of parameter input in an incompatible way  At FormatToBytesWritableMapper.java:[lines 374-380] |
|  |  Possible null pointer dereference in org.apache.phoenix.mapreduce.MultiHfileOutputFormat.configureIncrementalLoad(Job, List) due to return value of called method  Dereferenced at MultiHfileOutputFormat.java:org.apache.phoenix.mapreduce.MultiHfileOutputFormat.configureIncrementalLoad(Job, List) due to return value of called method  Dereferenced at MultiHfileOutputFormat.java:[line 691] |
|  |  Possible null pointer dereference in org.apache.phoenix.mapreduce.MultiHfileOutputFormat.getTableConfigurations(Configuration, String) due to return value of called method  Dereferenced at MultiHfileOutputFormat.java:org.apache.phoenix.mapreduce.MultiHfileOutputFormat.getTableConfigurations(Configuration, String) due to return value of called method  Dereferenced at MultiHfileOutputFormat.java:[line 348] |
|  |  org.apache.phoenix.mapreduce.bulkload.TargetTableRefFunctions.FROM_JSON isn't final but should be  At TargetTableRefFunctions.java:be  At TargetTableRefFunctions.java:[line 48] |
|  |  org.apache.phoenix.mapreduce.bulkload.TargetTableRefFunctions.LOGICAL_NAMES_TO_JSON isn't final but should be  At TargetTableRefFunctions.java:be  At TargetTableRefFunctions.java:[line 78] |
|  |  org.apache.phoenix.mapreduce.bulkload.TargetTableRefFunctions.NAMES_FROM_JSON isn't final but should be  At TargetTableRefFunctions.java:be  At TargetTableRefFunctions.java:[line 95] |
|  |  org.apache.phoenix.mapreduce.bulkload.TargetTableRefFunctions.NAMES_TO_JSON isn't final but should be  At TargetTableRefFunctions.java:be  At TargetTableRefFunctions.java:[line 61] |
|  |  org.apache.phoenix.mapreduce.bulkload.TargetTableRefFunctions.TO_JSON isn't final but should be  At TargetTableRefFunctions.java:be  At TargetTableRefFunctions.java:[line 35] |
|  |  qCol must be non-null but is marked as nullable  At IndexColumnNames.java:is marked as nullable  At IndexColumnNames.java:[line 161] |
|  |  Possible null pointer dereference in org.apache.phoenix.pherf.workload.mt.handlers.RendezvousingWorkHandler.onEvent(BaseLoadEventGenerator$TenantOperationEvent) due to return value of called method  Dereferenced at RendezvousingWorkHandler.java:org.apache.phoenix.pherf.workload.mt.handlers.RendezvousingWorkHandler.onEvent(BaseLoadEventGenerator$TenantOperationEvent) due to return value of called method  Dereferenced at RendezvousingWorkHandler.java:[line 76] |
|  |  Possible null pointer dereference in org.apache.phoenix.pherf.workload.mt.handlers.TenantOperationWorkHandler.onEvent(BaseLoadEventGenerator$TenantOperationEvent) due to return value of called method  Dereferenced at TenantOperationWorkHandler.java:org.apache.phoenix.pherf.workload.mt.handlers.TenantOperationWorkHandler.onEvent(BaseLoadEventGenerator$TenantOperationEvent) due to return value of called method  Dereferenced at TenantOperationWorkHandler.java:[line 65] |
|  |  input must be non-null but is marked as nullable  At IdleTimeOperationSupplier.java:is marked as nullable  At IdleTimeOperationSupplier.java:[lines 51-74] |
|  |  input must be non-null but is marked as nullable  At PreScenarioOperationSupplier.java:is marked as nullable  At PreScenarioOperationSupplier.java:[lines 51-85] |
|  |  input must be non-null but is marked as nullable  At QueryOperationSupplier.java:is marked as nullable  At QueryOperationSupplier.java:[lines 54-94] |
|  |  input must be non-null but is marked as nullable  At UpsertOperationSupplier.java:is marked as nullable  At UpsertOperationSupplier.java:[lines 58-161] |
|  |  input must be non-null but is marked as nullable  At UserDefinedOperationSupplier.java:is marked as nullable  At UserDefinedOperationSupplier.java:[lines 43-47] |
|  |  Method org.apache.phoenix.schema.ArgumentTypeMismatchException$1.apply(Class) overrides the nullness annotation of parameter input in an incompatible way  At ArgumentTypeMismatchException.java:annotation of parameter input in an incompatible way  At ArgumentTypeMismatchException.java:[line 51] |
|  |  Method org.apache.phoenix.schema.ArgumentTypeMismatchException$1.apply(Object) overrides the nullness annotation of parameter $L1 in an incompatible way  At ArgumentTypeMismatchException.java:annotation of parameter $L1 in an incompatible way  At ArgumentTypeMismatchException.java:[line 48] |
|  |  Dead store to metaProperties in org.apache.phoenix.schema.MetaDataClient.addColumn(PTable, List, ListMultimap, boolean, boolean, NamedTableNode, PTableType, boolean, List)  At MetaDataClient.java:org.apache.phoenix.schema.MetaDataClient.addColumn(PTable, List, ListMultimap, boolean, boolean, NamedTableNode, PTableType, boolean, List)  At MetaDataClient.java:[line 3896] |
|  |  Found reliance on default encoding in org.apache.phoenix.schema.MetaDataClient.addColumn(PTable, List, ListMultimap, boolean, boolean, NamedTableNode, PTableType, boolean, List):in org.apache.phoenix.schema.MetaDataClient.addColumn(PTable, List, ListMultimap, boolean, boolean, NamedTableNode, PTableType, boolean, List): String.getBytes()  At MetaDataClient.java:[line 4157] |
|  |  Possible null pointer dereference of colDef in org.apache.phoenix.schema.MetaDataClient.addColumn(PTable, List, ListMultimap, boolean, boolean, NamedTableNode, PTableType, boolean, List)  Dereferenced at MetaDataClient.java:colDef in org.apache.phoenix.schema.MetaDataClient.addColumn(PTable, List, ListMultimap, boolean, boolean, NamedTableNode, PTableType, boolean, List)  Dereferenced at MetaDataClient.java:[line 3976] |
|  |  org.apache.phoenix.schema.MetaDataClient.addColumn(PTable, List, ListMultimap, boolean, boolean, NamedTableNode, PTableType, boolean, List) may fail to close PreparedStatement  At MetaDataClient.java:NamedTableNode, PTableType, boolean, List) may fail to close PreparedStatement  At MetaDataClient.java:[line 4133] |
|  |  Nullcheck of newIndexes at line 190 of value previously dereferenced in org.apache.phoenix.schema.PMetaDataImpl.removeTable(PName, String, String, long)  At PMetaDataImpl.java:190 of value previously dereferenced in org.apache.phoenix.schema.PMetaDataImpl.removeTable(PName, String, String, long)  At PMetaDataImpl.java:[line 190] |
|  |  Method org.apache.phoenix.util.QueryUtil$2.apply(Object) overrides the nullness annotation of parameter $L1 in an incompatible way  At QueryUtil.java:annotation of parameter $L1 in an incompatible way  At QueryUtil.java:[line 220] |
|  |  Method org.apache.phoenix.util.QueryUtil$2.apply(String) overrides the nullness annotation of parameter columnName in an incompatible way  At QueryUtil.java:annotation of parameter columnName in an incompatible way  At QueryUtil.java:[line 224] |
|  |  Method org.apache.phoenix.util.csv.CsvUpsertExecutor$ArrayDatatypeConversionFunction.apply(Object) overrides the nullness annotation of parameter $L1 in an incompatible way  At CsvUpsertExecutor.java:annotation of parameter $L1 in an incompatible way  At CsvUpsertExecutor.java:[line 217] |
|  |  Method org.apache.phoenix.util.csv.CsvUpsertExecutor$ArrayDatatypeConversionFunction.apply(String) overrides the nullness annotation of parameter input in an incompatible way  At CsvUpsertExecutor.java:annotation of parameter input in an incompatible way  At CsvUpsertExecutor.java:[lines 229-231] |
|  |  Method org.apache.phoenix.util.csv.CsvUpsertExecutor$SimpleDatatypeConversionFunction.apply(Object) overrides the nullness annotation of parameter $L1 in an incompatible way  At CsvUpsertExecutor.java:annotation of parameter $L1 in an incompatible way  At CsvUpsertExecutor.java:[line 123] |
|  |  Method org.apache.phoenix.util.csv.CsvUpsertExecutor$SimpleDatatypeConversionFunction.apply(String) overrides the nullness annotation of parameter input in an incompatible way  At CsvUpsertExecutor.java:annotation of parameter input in an incompatible way  At CsvUpsertExecutor.java:[lines 168-210] |
|  |  Method org.apache.phoenix.util.json.JsonUpsertExecutor$ArrayDatatypeConversionFunction.apply(Object) overrides the nullness annotation of parameter input in an incompatible way  At JsonUpsertExecutor.java:annotation of parameter input in an incompatible way  At JsonUpsertExecutor.java:[lines 251-253] |
|  |  Method org.apache.phoenix.util.json.JsonUpsertExecutor$SimpleDatatypeConversionFunction.apply(Object) overrides the nullness annotation of parameter input in an incompatible way  At JsonUpsertExecutor.java:annotation of parameter input in an incompatible way  At JsonUpsertExecutor.java:[lines 186-232] |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1317/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1317 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile spotbugs hbaseanti checkstyle |
| uname | Linux bd2b89842ded 4.15.0-147-generic #151-Ubuntu SMP Fri Jun 18 19:21:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / 3da4ed2 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1317/1/artifact/yetus-general-check/output/diff-checkstyle-root.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1317/1/artifact/yetus-general-check/output/whitespace-tabs.txt |
| spotbugs | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1317/1/artifact/yetus-general-check/output/new-spotbugs-phoenix-core.html |
| spotbugs | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1317/1/artifact/yetus-general-check/output/new-spotbugs-phoenix-pherf.html |
| spotbugs | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1317/1/artifact/yetus-general-check/output/new-spotbugs-root.html |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1317/1/artifact/yetus-general-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1317/1/testReport/ |
| Max. process+thread count | 5009 (vs. ulimit of 30000) |
| modules | C: phoenix-core phoenix-pherf phoenix-server . U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1317/1/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: phoenix-core/pom.xml [hunk 1]
```diff
@@ -282,34 +282,35 @@
   </dependency>
 
     <!-- Transaction dependencies -->
+    <!--
```
Comment (stoty) [inline phoenix-core/pom.xml:285]: Nit: delete instead of commenting out

File: phoenix-core/pom.xml [hunk 2]
```diff
@@ -282,34 +282,35 @@
...
+    -->
     <dependency>
       <groupId>org.apache.tephra</groupId>
-      <artifactId>tephra-core</artifactId>
+      <artifactId>tephra-core-shaded</artifactId>
     </dependency>
     <dependency>
       <groupId>org.apache.tephra</groupId>
       <artifactId>tephra-core</artifactId>
+      <version>${tephra.version}</version>
```
Comment (stoty) [inline phoenix-core/pom.xml:298]: The versions should be defined in the top project dependencyManagement section

File: phoenix-pherf/pom.xml [hunk 3]
```diff
@@ -118,6 +123,7 @@
 		<dependency>
 			<groupId>org.apache.tephra</groupId>
 			<artifactId>tephra-core</artifactId>
+			<version>${tephra.version}</version>
```
Comment (stoty) [inline phoenix-pherf/pom.xml:126]: This should already be dependencyMananged in the top pom

File: phoenix-core/pom.xml [hunk 4]
```diff
@@ -282,34 +282,35 @@
...
       <type>test-jar</type>
       <scope>test</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.tephra</groupId>
       <artifactId>tephra-hbase-compat-${tephra.hbase.compat.version}</artifactId>
+      <version>${tephra.version}</version>
```
Comment (stoty) [inline phoenix-core/pom.xml:305]: The versions should be defined in the top project dependencyManagement section

File: phoenix-server/pom.xml [hunk 5]
```diff
@@ -173,6 +173,7 @@
     <dependency>
       <groupId>org.apache.tephra</groupId>
       <artifactId>tephra-core</artifactId>
+      <version>${tephra.version}</version>
```
Comment (stoty) [inline phoenix-server/pom.xml:176]: The versions should be defined in the top project dependencyManagement section

(Why do we even have test dependencies in phoenix-server ? - that's a question for another day)

**Review by stoty — APPROVED** [review]
Looks good.
Could you please clean up the version handling in the poms before commit ?

**Review by virajjasani — APPROVED** [review]
+1 overall, worth fixing the above comments

**Comment by gjacoby126** [comment]
Fixed the versions (which revealed some extra unshaded twill dependencies in UpdateStatisticsTool) and reran tests locally which passed. Merging and then will backport to 4.16. Thanks for the reviews @stoty and @virajjasani !

**Pull request merged**